cpp/include/raft/neighbors/detail/ivf_pq_build.cuh

```diff
@@ -727,6 +662,12 @@ __launch_bounds__(BlockSize) __global__ void process_and_fill_codes_kernel(
   const uint32_t pq_len     = pq_centers.extent(1);
   const uint32_t pq_dim     = new_vectors.extent(1) / pq_len;
 
+  auto pq_extents = make_list_extents<uint32_t>(out_ix + 1, PqBits, pq_dim);
+  auto pq_extents_vectorized =
+    make_extents<uint32_t>(pq_extents.extent(0), pq_extents.extent(1), pq_extents.extent(2));
+  auto pq_dataset = make_mdspan<pq_vec_t, uint32_t, row_major, false, true>(
```

Comment:
That's a cuda kernel code Corey :) Also this is a really an implementation detail that is hard to workaround.  `pq_vec_t` is in fact a cuda simd type, and the whole extents/mdspan exercise is only needed because we don't have submdspan yet (which I miss a lot!).

Reply:
> That's a cuda kernel code Corey :) 

Doh! I see that now- the way github presents the diff here somewhat obscures the function signature and I'm more used to seeing the mdspan in non-device code. Yeah this is fine- I was also somewhat responding to Tamas' earlier question as to whether we should prefer to use mdspan<shared_pr<>> or just use std::vector and in those cases I'd prefer to use std::vector.

Reply:
Maybe a sub mdspan is something we should prioritize after things calm down. We have talked about it for awhile.